src/sentry/api/authentication.py

```diff
@@ -292,37 +341,59 @@ def accepts_auth(self, auth: list[bytes]) -> bool:
         return not token_str.startswith(SENTRY_ORG_AUTH_TOKEN_PREFIX)
 
     def authenticate_token(self, request: Request, token_str: str) -> tuple[Any, Any]:
-        token = SystemToken.from_request(request, token_str)
-        try:
-            token = (
-                token
-                or ApiToken.objects.filter(token=token_str)
-                .select_related("user", "application")
-                .get()
-            )
-        except ApiToken.DoesNotExist:
+        user: AnonymousUser | RpcUser | None = AnonymousUser()
+
+        token: SystemToken | ApiTokenReplica | ApiToken | None = SystemToken.from_request(
+            request, token_str
+        )
+        application_is_active = True
+
+        if not token:
+            if SiloMode.get_current_mode() == SiloMode.REGION:
+                atr = token = ApiTokenReplica.objects.filter(token=token_str).last()
+                if not atr:
+                    raise AuthenticationFailed("Invalid token")
+                user = user_service.get_user(user_id=atr.user_id)
+                application_is_active = atr.application_is_active
+            else:
+                try:
+                    at = token = (
+                        ApiToken.objects.filter(token=token_str)
+                        .select_related("user", "application")
+                        .get()
+                    )
+                except ApiToken.DoesNotExist:
+                    raise AuthenticationFailed("Invalid token")
+                user = at.user
+                application_is_active = at.application is None or at.application.is_active
```

Comment:
If the token has no `.application` shouldn't `application_is_active` be False?  I can see how the end result is correct based on the check on 381, but this being true when the application doesn't exist feels like a footgun for future us.